plutus-metatheory/src/Builtin.lagda.md

```diff
@@ -144,7 +144,7 @@ This is defined in its own module so that these definitions are not exported.
 
     infix 10 _[_
     _[_ : (n : ℕ) →  Args n  ×  n ⊢♯ → Sig
```

Comment:
this is a bit mysterious? What's going on here?

Reply:
It's part of the syntactic sugar for defining the signatures of builtins with a nice syntax. A signature is composed of three things, a number, a list of types and a result type ,and we want to write them like
```
  n [ t1 , t2 , t3 ]-> result 
```
so I defined  
 * `_[_` to take the number and the list with the result type
 * `_,_` to take a type and (list and a result type) and add one element the list
  * `_]->_` to take a type and result, make a singleton with the type  list a return with the result type.
  So, when you write
```
  n [ t1 , t2 , t3 ]-> result 
```
you are actually writing
```
 n [  (t1 , (t2 , ( t3 ]-> result))) 
```

Reply:
ah! thanks for explaining the signatures look very nice! Maybe worth a comment here.

Reply:
Added some explanations (and reimplemented to construct the lists without having to reverse it).